datamonster_api/lib/datamonster.py

```diff
@@ -158,16 +158,12 @@ def get_datasources(self, query=None, company=None):
 
     def get_datasource_by_name(self, name):
         """Given a name, try to find a datasource"""
-        result = list(self.get_datasources(query=name))
-        if not len(result):
-            raise DataMonsterError(
-                "Did not find a data source matching the name {!r}".format(name)
-            )
-        if len(result) > 1:
-            raise DataMonsterError(
-                "Found {} data sources matching the name {!r}".format(len(result), name)
-            )
-        return result.pop()
+        for ds in self.get_datasources(query=name):
+            if ds.name.lower() == name.lower():
+                return ds
+        raise DataMonsterError(
+            "Did not find a data source matching the name {!r}".format(name)
+        )
```

Comment:
I like it. The only issue I see will be if there are two with the same name, but I don't think that's a problem to solve here.